src/coloring/acyclic_coloring.jl

```diff
@@ -10,29 +10,30 @@ is a collection of trees—and hence is acyclic.
 Reference: Gebremedhin AH, Manne F, Pothen A. **New Acyclic and Star Coloring Algorithms with Application to Computing Hessians**
 """
 function color_graph(g::LightGraphs.AbstractGraph, ::AcyclicColoring)
-
     color = zeros(Int, nv(g))
-    forbidden_colors = zeros(Int, nv(g))
+    set = DisjointSets{Int}([])
+
+    first_visit_to_tree = Array{Tuple{Int, Int}, 1}()
+    first_neighbor = Array{Tuple{Int, Int}, 1}()
 
-    set = DisjointSets{LightGraphs.Edge}([])
+    init_array!(first_visit_to_tree, ne(g))
+    init_array!(first_neighbor, ne(g))
 
-    first_visit_to_tree = Array{Tuple{Int, Int}, 1}(undef, ne(g))
-    first_neighbor = Array{Tuple{Int, Int}, 1}(undef, nv(g))
+    forbidden_colors = zeros(Int, nv(g))
 
     for v in vertices(g)
-        #enforces the first condition of acyclic coloring
         for w in outneighbors(g, v)
-            if color[w] != 0
+            if color[w]!=0
                 forbidden_colors[color[w]] = v
             end
         end
-        #enforces the second condition of acyclic coloring
+
         for w in outneighbors(g, v)
-            if color[w] != 0 #colored neighbor
+            if color[w]!=0
```

Comment:
Keep spaces around operators

```suggestion
            if color[w] != 0
```